packages/core/helpers/ver/4.24.9.ts

```diff
@@ -1,17 +1,50 @@
 import _ from 'lodash'
+import { AnyVisualization } from '../../types/Visualization'
+import { VizFilter } from '../../types/VizFilter'
+import versionNeedsUpdate from './versionNeedsUpdate'
 
-const patchSingleStateZoom = newConfig => {
-  // Map zooming is a default feature created for world maps
-  // Since we're turning it on for single state maps lets default it off for opting in.
-  if (newConfig.type === 'map' && newConfig.general.geoType === 'single-state') {
-    newConfig.general.allowMapZoom = false
+/**
+ * Updates the configuration object to disable map zooming for single state maps in a dashboard.
+ * when the config is below version 4.24.9
+ */
+const patchSingleStateZoom = config => {
+  if (config.type === 'dashboard') {
+    Object.values(config?.visualizations || {}).forEach((viz: AnyVisualization) => {
+      if (viz.type === 'map' && viz.general?.geoType === 'single-state') {
+        // if the version is less that 4.24.7 then allowMapZoom should be false
+        let allowZoom = versionNeedsUpdate(config.version, '4.24.9') ? false : viz.general.allowMapZoom
+        viz.general.allowMapZoom = allowZoom
+      }
+    })
+  }
+  if (config.type === 'map' && config.general?.geoType === 'single-state') {
+    // if the version is less that 4.24.7 then allowMapZoom should be false
+    let allowZoom = versionNeedsUpdate(config.version, '4.24.9') ? false : config.general.allowMapZoom
+    config.general.allowMapZoom = allowZoom
+  }
+}
+
+const doesNotHaveIds = (filters: VizFilter[]) => filters?.some(f => !f.id)
+
+export const addIdsToVisFilters = config => {
```

Comment:
can you remove this function, instead of commenting it out. it will be easier to rebase if the function was never there vs having to uncomment.

Reply:
and then make sure the commits get squash so you don't have a commit where it was removed.